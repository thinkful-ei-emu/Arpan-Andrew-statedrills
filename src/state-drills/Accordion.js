import React from 'react';

export default class Accordian extends React.Component{

    static defaultProps = {
        sections: []
    }

    state = {
      currentActive: null
    }

    handleClick = e => {      
      this.setState({
        currentActive: e.target.id
      });
    }  

    render(){
      const keys = this.props.sections.map((key, index) => {
        return <li key={index}>
          <button id={index.toString()} onClick={this.handleClick}>{key.title}</button>
          {(this.state.currentActive === index.toString()) && <p>{key.content}</p>}
        </li>
      });

      return (
        <ul>
          {keys}
        </ul>
      )
    }
}