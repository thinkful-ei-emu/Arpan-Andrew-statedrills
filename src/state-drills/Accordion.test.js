import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion';

describe ('Accordion', () => {

    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]
      // e.g. <Accordion sections={sections} />


      it('Renders without errors', () =>{
          const div = document.createElement('div');
          ReactDOM.render(<Accordion />, div);
          ReactDOM.unmountComponentAtNode(div);
      })

      it('Renders empty li when sections not supplied', () => {
          const wrapper = shallow(<Accordion />)
          expect(toJson(wrapper)).toMatchSnapshot()
      })

      it('renders no sections as active by default', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })

      it('component opens a clicked section ', () => {
        const wrapper = mount(<Accordion sections={sections} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })

      it('component only opens the last section ', () => {
        const wrapper = mount(<Accordion sections={sections} />)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})