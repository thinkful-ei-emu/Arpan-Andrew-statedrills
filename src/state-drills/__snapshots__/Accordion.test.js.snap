// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Renders empty li when sections not supplied 1`] = `<ul />`;

exports[`Accordion component only opens the last section  1`] = `
<Accordian
  sections={
    Array [
      Object {
        "content": "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
        "title": "Section 1",
      },
      Object {
        "content": "Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!",
        "title": "Section 2",
      },
      Object {
        "content": "Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?",
        "title": "Section 3",
      },
    ]
  }
>
  <ul>
    <li
      key="0"
    >
      <button
        id="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        id="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        id="2"
        onClick={[Function]}
      >
        Section 3
      </button>
      <p>
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</Accordian>
`;

exports[`Accordion component opens a clicked section  1`] = `
<Accordian
  sections={
    Array [
      Object {
        "content": "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
        "title": "Section 1",
      },
      Object {
        "content": "Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!",
        "title": "Section 2",
      },
      Object {
        "content": "Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?",
        "title": "Section 3",
      },
    ]
  }
>
  <ul>
    <li
      key="0"
    >
      <button
        id="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        id="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        id="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</Accordian>
`;

exports[`Accordion renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      id="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      id="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      id="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
